tdms/include/surface_phasors.h

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file surface_phasors.h
+ * @brief
+ */
+#pragma once
+
+#include "arrays.h"
+#include "field.h"
+#include "grid_labels.h"
+
+class SurfacePhasors {
+private:
+  int **surface_vertices = nullptr;//< Pointer to the vertices on the surface
+  int n_surface_vertices = 0;      //< Number of vertices on the surface
+
+  mxArray *vertex_list = nullptr;         //< List of vertices
+  double **vertex_list_data_ptr = nullptr;//< Buffer to place vertex data into MATLAB array
+
+  mxArray *mx_surface_amplitudes = nullptr;//< Complex surface amplitudes
+  int f_ex_vector_size = 0; //< Number of elements in the frequency extraction vector
+
+  /* Storage for real and imag parts of mx_surface_amplitudes (these can be f_ex_vector_size * n_surface_vertices arrays of FullFieldSnapshots when MATLAB is removed!)
+
+  Arrays are index by [frequency_index][field component][vertex_id/number]. Frequency index corresponds to the frequencies at which the user has requested we extract the amplitudes.
+  */
+  double ***surface_EHr = nullptr, ***surface_EHi = nullptr;
+
+public:
+  SurfacePhasors() = default;
+  /**
+   * @brief Construct a new Surface Phasors object, using the set_from_matlab_array() method
+   */
+  SurfacePhasors(mxArray *mx_surface_vertices, int _f_ex_vector_size);
+  /**
+   * @brief Sets the surface_vertices pointer and number of surface vertices tracker from the MATLAB array passed in
+   *
+   * @param mx_surface_vertices MATLAB array containing the vertex information
+   * @param _f_ex_vector_size The FrequencyExtractionVector size, which we need to reserve sufficient memory
+   */
+  void set_from_matlab_array(mxArray *mx_surface_vertices, int f_ex_vector_size);
+
+  /**
+   * @brief Zeros the surface_EH{r,i} arrays
+   */
+  void zero_surface_EH() {
+    for (int k = 0; k < f_ex_vector_size; k++) {
+      for(int j = 0; j < 6; j++) {
```

Comment:
```suggestion
      for(int j = 0; j < 6; j++) { // 6 field components: Ex, y, z, and Hx, y, z
```
Just because I didn't immediately remember your indexing comment.